components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java

```diff
@@ -217,4 +227,74 @@ private void unsetCurrentSessionIdThreadLocal() {
+            throws IOException, ServletException {
+
+        try {
+            TenantManager tenantManager = AuthenticationValveServiceHolder.getInstance().getRealmService()
+                    .getTenantManager();
+            if (tenantDomain != null && !tenantManager.isTenantActive(IdentityTenantUtil.getTenantId(tenantDomain))) {
+                handleInvalidTenantDomainErrorResponse(request, response, HttpServletResponse.SC_NOT_FOUND,
+                        tenantDomain);
+                return false;
+            }
+        } catch (UserStoreException ex) {
+            if (log.isDebugEnabled()) {
+                log.debug("Error occurred while validating tenant domain.", ex);
+            }
+            handleInvalidTenantDomainErrorResponse(request, response, HttpServletResponse.SC_NOT_FOUND, tenantDomain);
+            return false;
+        } catch (IdentityRuntimeException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Error occurred while validating tenant domain.", e);
+            }
+            String INVALID_TENANT_DOMAIN = "Invalid tenant domain";
+            if (!StringUtils.isBlank(e.getMessage()) && e.getMessage().contains(INVALID_TENANT_DOMAIN)) {
+                handleInvalidTenantDomainErrorResponse(request, response, HttpServletResponse.SC_NOT_FOUND,
+                        tenantDomain);
+            } else {
+                handleRuntimeErrorResponse(response, HttpServletResponse.SC_INTERNAL_SERVER_ERROR, tenantDomain);
+            }
+            return false;
+        }
+        return true;
+    }
+
+    private void handleInvalidTenantDomainErrorResponse(Request request, Response response, int error,
+                                                        String tenantDomain) throws
+            IOException {
+
+        String requestContentType = request.getContentType();
+        response.setStatus(error);
+        response.setCharacterEncoding("UTF-8");
+        if (StringUtils.equalsIgnoreCase("application/json", requestContentType)) {
+            response.setContentType("application/json");
+            JsonObject errorResponse = new JsonObject();
+            String errorMsg = "invalid tenant domain : " + tenantDomain;
+            errorResponse.addProperty("code", error);
+            errorResponse.addProperty("message", errorMsg);
+            errorResponse.addProperty("description", errorMsg);
+            response.getWriter().print(errorResponse.toString());
+        } else {
+            response.setContentType("text/html");
+            String errorPage = AuthenticationValveDataHolder.getInstance().getInvalidTenantDomainErrorPage();
+            errorPage = errorPage.replace("$tenant.name", tenantDomain);
+            response.getWriter().print(errorPage);
+        }
+    }
+
+    private void handleRuntimeErrorResponse(Response response, int error, String tenantDomain) throws
+            IOException {
+
+        response.setContentType("application/json");
```

Comment:
Don't we need to handle html content type here as well?

Reply:
Fixed in 15e0d61